src/windows_utils.rs

```diff
@@ -1,222 +1,177 @@
-    let str_registry_path_2 = CString::new("Software\\Classes\\*\\shell\\Neovide").unwrap();
     unsafe {
-        let s1 = RegDeleteTreeA(HKEY_CURRENT_USER, str_registry_path_1.as_ptr());
-        let s2 = RegDeleteTreeA(HKEY_CURRENT_USER, str_registry_path_2.as_ptr());
-        s1 == 0 && s2 == 0
+        RegDeleteTreeA(HKEY_CURRENT_USER, REGISTRY_PATH_DIRECTORY).0 == 0
+            && RegDeleteTreeA(HKEY_CURRENT_USER, REGISTRY_PATH_FOLDER).0 == 0
     }
 }
 
 fn register_rightclick_directory() -> bool {
-    let neovide_path = get_binary_path();
-    let mut registry_key: HKEY = null_mut();
-    let str_registry_path =
-        CString::new("Software\\Classes\\Directory\\Background\\shell\\Neovide").unwrap();
-    let str_registry_command_path =
-        CString::new("Software\\Classes\\Directory\\Background\\shell\\Neovide\\command").unwrap();
-    let str_icon = CString::new("Icon").unwrap();
-    let str_command = CString::new(format!("{} \"%V\"", neovide_path).as_bytes()).unwrap();
-    let str_description = CString::new("Open with Neovide").unwrap();
-    let str_neovide_path = CString::new(neovide_path.as_bytes()).unwrap();
+    let mut registry_key = HKEY::default();
+
+    let neovide_path = get_neovide_path();
+    let neovide_description = "Open with Neovide";
+    let neovide_command = format!("{} \"%V\"", neovide_path);
+
     unsafe {
         if RegCreateKeyExA(
             HKEY_CURRENT_USER,
-            str_registry_path.as_ptr(),
+            REGISTRY_PATH_DIRECTORY,
             0,
-            null_mut(),
+            PCSTR::null(),
             REG_OPTION_NON_VOLATILE,
-            KEY_WRITE,
-            null_mut(),
+            KEY_READ | KEY_WRITE,
+            None,
             &mut registry_key,
-            null_mut(),
-        ) != 0
+            None,
+        )
+        .0 != 0
         {
-            RegCloseKey(registry_key);
+            let _ = RegCloseKey(registry_key);
             return false;
         }
-        let registry_values = [
-            (
-                null(),
-                REG_SZ,
-                str_description.as_ptr() as *const u8,
-                str_description.to_bytes().len() + 1,
-            ),
-            (
-                str_icon.as_ptr(),
-                REG_SZ,
-                str_neovide_path.as_ptr() as *const u8,
-                str_neovide_path.to_bytes().len() + 1,
-            ),
-        ];
-        for &(key, keytype, value_ptr, size_in_bytes) in &registry_values {
-            RegSetValueExA(
-                registry_key,
-                key,
-                0,
-                keytype,
-                value_ptr,
-                size_in_bytes as u32,
-            );
-        }
-        RegCloseKey(registry_key);
+
+        let _ = RegSetValueExA(
+            registry_key,
+            PCSTR::null(),
+            0,
+            REG_SZ,
+            Some(neovide_description.as_bytes()),
+        );
```

Comment:
I think we should return a failure from this function instead of ignoring the error if the setting the value fails. NOTE: The key still needs to be closed in that case.